java/src/jmri/jmrit/display/PositionableLabel.java

```diff
@@ -507,10 +508,12 @@ public boolean isRotated() {
     }
 
     public void updateIcon(NamedIcon s) {
-        _namedIcon = s;
-        super.setIcon(_namedIcon);
-        updateSize();
-        repaint();
+        ThreadingUtil.runOnLayoutEventually(() -> {
```

Comment:
It would be better to use `ThreadingUtil.runOnGUIEventually()`. The layout thread is currently the same as the GUI thread, but in a distant future, they could be separated.